# Space Titanic dataset analyse

Import required packages

In [2]:
import pandas as pd
import matplotlib.pyplot as plt
import os

# Load the dataset

In [4]:
filename = "train.csv"
df = pd.read_csv(os.path.join("data", filename))

# Inspect the dataset

In [9]:
df.head()

PassengerId HomePlanet CryoSleep  Cabin  Destination   Age    VIP  \
0     0001_01     Europa     False  B/0/P  TRAPPIST-1e  39.0  False   
1     0002_01      Earth     False  F/0/S  TRAPPIST-1e  24.0  False   
2     0003_01     Europa     False  A/0/S  TRAPPIST-1e  58.0   True   
3     0003_02     Europa     False  A/0/S  TRAPPIST-1e  33.0  False   
4     0004_01      Earth     False  F/1/S  TRAPPIST-1e  16.0  False   

   RoomService  FoodCourt  ShoppingMall     Spa  VRDeck               Name  \
0          0.0        0.0           0.0     0.0     0.0    Maham Ofracculy   
1        109.0        9.0          25.0   549.0    44.0       Juanna Vines   
2         43.0     3576.0           0.0  6715.0    49.0      Altark Susent   
3          0.0     1283.0         371.0  3329.0   193.0       Solam Susent   
4        303.0       70.0         151.0   565.0     2.0  Willy Santantines   

   Transported  
0        False  
1         True  
2        False  
3        False  
4         True

In [24]:
print(f"Shape: {df.shape}")
print()
print(f"Datatypes:\n{df.dtypes}")

print()

Shape: (8693, 14)

Datatypes:
PassengerId      object
HomePlanet       object
CryoSleep        object
Cabin            object
Destination      object
Age             float64
VIP              object
RoomService     float64
FoodCourt       float64
ShoppingMall    float64
Spa             float64
VRDeck          float64
Name             object
Transported        bool
dtype: object



## Inspect missing values

In [19]:
df[df.isna().sum(axis=1) > 0]

PassengerId HomePlanet CryoSleep     Cabin  Destination   Age    VIP  \
7        0006_02      Earth      True     G/0/S  TRAPPIST-1e  28.0  False   
10       0008_02     Europa      True     B/1/P  TRAPPIST-1e  34.0  False   
15       0012_01      Earth     False       NaN  TRAPPIST-1e  31.0  False   
16       0014_01       Mars     False     F/3/P  55 Cancri e  27.0  False   
23       0020_03      Earth      True     E/0/S  55 Cancri e  29.0  False   
...          ...        ...       ...       ...          ...   ...    ...   
8667     9250_01     Europa     False   E/597/P  TRAPPIST-1e  29.0  False   
8674     9257_01        NaN     False  F/1892/P  TRAPPIST-1e  13.0  False   
8675     9259_01      Earth       NaN  F/1893/P  TRAPPIST-1e  44.0  False   
8684     9274_01        NaN      True  G/1508/P  TRAPPIST-1e  23.0  False   
8687     9275_03     Europa       NaN    A/97/P  TRAPPIST-1e  30.0  False   

      RoomService  FoodCourt  ShoppingMall   Spa  VRDeck                Name  \
7             0.0        0.0           0.0   0.0     NaN  Candra Jacostaffey   
10            0.0        0.0           NaN   0.0     0.0      Altardr Flatic   
15           32.0        0.0         876.0   0.0     0.0       Justie Pooles   
16         1286.0      122.0           NaN   0.0     0.0         Flats Eccle   
23            0.0        0.0           NaN   0.0     0.0  Mollen Mcfaddennon   
...           ...        ...           ...   ...     ...                 ...   
8667          0.0     2972.0           NaN  28.0   188.0    Chain Reedectied   
8674         39.0        0.0        1085.0  24.0     0.0          Ties Apple   
8675       1030.0     1015.0           0.0  11.0     NaN     Annah Gilleyons   
8684          0.0        0.0           0.0   0.0     0.0     Chelsa Bullisey   
8687          0.0     3208.0           0.0   2.0   330.0     Atlasym Conable   

      Transported  
7            True  
10           True  
15          False  
16          False  
23          False  
...           ...  
8667         True  
8674        False  
8675         True  
8684         True  
8687         True  

[2087 rows x 14 columns]

In [20]:
df[df.isna().sum(axis=1) > 0].isna().sum(axis=1).value_counts()

1    1867
2     203
3      17
Name: count, dtype: int64

In [16]:
df[df["HomePlanet"].isna()].isna().sum()

PassengerId       0
HomePlanet      201
CryoSleep         2
Cabin             6
Destination       4
Age               2
VIP               3
RoomService       6
FoodCourt         2
ShoppingMall      7
Spa               3
VRDeck            1
Name              7
Transported       0
dtype: int64

In [32]:
df["CryoSleep"].value_counts()

CryoSleep
False    5439
True     3037
Name: count, dtype: int64

In [36]:
df["CryoSleep"].dtype

dtype('O')

In [40]:
df[df["CryoSleep"]==True].describe()

Age  RoomService  FoodCourt  ShoppingMall     Spa  VRDeck
count  2955.000000       2969.0     2967.0        2941.0  2972.0  2975.0
mean     27.405415          0.0        0.0           0.0     0.0     0.0
std      15.080469          0.0        0.0           0.0     0.0     0.0
min       0.000000          0.0        0.0           0.0     0.0     0.0
25%      18.000000          0.0        0.0           0.0     0.0     0.0
50%      26.000000          0.0        0.0           0.0     0.0     0.0
75%      37.000000          0.0        0.0           0.0     0.0     0.0
max      78.000000          0.0        0.0           0.0     0.0     0.0

Conclusions:
 - Dataset contains 8693 rows described by 14 attributes
 - Columns are either floats or strings. Only target column is bool.
 - Almost all columns contain around 200 missing values. The majority of rows have 0 or 1 missing values.
    - 1867 rows have one missing value. Because of large percentage of those rows in dataset, some imputation will be needed.
    - Only 203 rows have 2 missing values. Only 17 rows have 3 missing values. There is no row with more than 3 missing values.
 - Column 'Transported' is target, that we have to predict.
 - 'Passenger_id' column contains two numbers: first indicates a group (number of ticket?) the passenger is travelling with. Second number is their number within the group.
 - 'CryoSleep' columns indicate whether passenger is confied to his room. Those people don't spend any money - this information may be helpful for data imputing.
 - VIP passenger has paid for special services
 - Last name of passenger may indicate, if passengers in group are a family.
 

Things to do during feature engineering and data cleaning phase:
 - impute missing values or remove rows containing them.
 - divide columns PassengerId and Cabin into more atomic attributes
 - bucketize 'Age' and expenses columns
 - Create new features: 'is_travelling_alone', 'is_travelling_with_family'